spring-cloud-generator/scripts/generate-library-list.sh

```diff
@@ -23,16 +23,21 @@ cd ${SPRING_GENERATOR_DIR}
 filename=${SPRING_GENERATOR_DIR}/scripts/resources/library_list.txt
 echo "# api_shortname, googleapis-folder, distribution_name:version, monorepo_folder" > "$filename"
 
-# loop through folders
+# loop through configs for the monorepo (google-cloud-java)
 count=0
-for d in ./google-cloud-java/*java-*/; do
-  # parse variables from .repo-metadata.json
-  language=$(cat $d/.repo-metadata.json | jq -r .language)
-  api_shortname=$(cat $d/.repo-metadata.json | jq -r .api_shortname)
-  distribution_name=$(cat $d/.repo-metadata.json | jq -r .distribution_name)
-  library_type=$(cat $d/.repo-metadata.json | jq -r .library_type)
-  release_level=$(cat $d/.repo-metadata.json | jq -r .release_level)
-  monorepo_folder=$(basename $d)
+configs=$(yq e '.libraries[]' ./google-cloud-java/generate_config.yaml)
+for config_str in $configs; do
+  # parse variables from generation_config.yaml
+  config=$(echo "$config_str" | yq e '.') # Parse config_str back to YAML format
+  unique_module_name=$(echo "$config" | jq -r '.library_name // .api_shortname')
+  # Determine distribution_name
+  distribution_name=$(echo "$config" | jq -r '.distribution_name' // "")
+  if [ -z "$distribution_name" ]; then
+    distribution_name="com.google.cloud:google-cloud-${unique_module_name}"
```

Comment:
Did we get the logic from [here](https://github.com/googleapis/sdk-platform-java/blob/main/library_generation/utils/utilities.py#L208-L212)? It will not work for non-cloud apis, but I think we are not publishing SpringCodeGen for them anyway, so we should be fine.

Reply:
Yep, that's correct. And yeah I assumed as much, but I've added an explicit comment about that now